proto/genesis/v1beta/querier.proto

```diff
@@ -39,6 +41,22 @@ message QueryPendingProposalsResponse {
     repeated Proposal proposals = 2;
 }
 
+message QueryApprovedProposalsRequest {
```

Comment:
What do you think about combining all query proposals into one request with a `type=proposal-type` filter in the request to make this API simpler to use?

`proposal-type` can be one of  _pending_, _approved_, _rejected_.

Reply:
I did this because I made the assumption that you rarely want to query all the proposals whatever the state is. It's easier to implement also since we group proposals in the store by their state.
If you think querying all proposals for a chain should be a feature, don't hesitate to raise an issue 👍